### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by pan-** [ARMmbed/connectedhomeip#60] [General]
We do have a set of engineering goals to solve to complete this project but not yet a roadmap: 

- POC: 
   - Platform abstractions:
      - Event/RTOS
      - BLE
      - WiFi station
      - Storage/configuration
      - Time support
   - Mbed integration
      - IP connectivity 
      - Mbed TLS 
   - Applications:
     - Shell application
     - Lock example 
- Documentation
     - Internal reference (build system, platform abstraction notes, test integration description)
     - User reference: 
       - Hardware requirement
       - Software integration
       - Configuration option
       - Example of integration for applications living outside the CHIP source tree. - 
- Quality 
   - Testing of platform abstraction implementation
     - Unit tests ?
     - Integration tests 
   - QEMU integration
   - Cirque integration 
   - Integration of CHIP CI with Mbed CI to run tests on boards
- Developer experience 
  - Docker env 
  - VScode experience 
     - Build 
     - Flash
     - Debug

Is there anything I'm missing ? 
In my opinion we should continue to push on the POC to have a base we can build on. Once the platform abstractions are complete we can expand on documentation and quality. 

As a timeline of events I was thinking the following: 

Sprint 4 -> Completed IP connectivity integration 
Sprint 5 -> Shell application + upstream of mbed port 
Sprint 6 -> Lock application upstreamed. 
Sprint 8 -> Integration and unit test complete 
Sprint 9 -> Documentation complete 
Sprint 12 -> Complete work package 

There's some activities that are not well understood yet such as everything related to testing. I propose we start these investigation activities on this topic in Sprint 3 to clarify what is delivered, when we start working on the topic and when we deliver. 

What is your view on this @lukgni @ATmobica @fkjagodzinski @tymoteuszblochmobica ?

From a project management standpoint, I'd like us to collectively fill the backlog and add relevant ticket to the right [project](https://github.com/ARMmbed/connectedhomeip/projects). Labels can help us navigating through the list of issues more easily too.  
 
We can open topic on specific items we want to discuss bellow.

**Comment by ATmobica** [comment]
LGTM.  I agree that we should take a look at things related to testing.  It can be quite a complex topic that may take more than one sprint. I think we should also investigate the debugging and flashing issues, discuss the best approach.
